Classes/Extensions/DDFileLogger+Buffering.m

```diff
@@ -37,67 +37,13 @@ static NSUInteger DDGetDefaultBufferSizeBytesMax(BOOL max) {
     return max ? kDDMaxBufferSize : kDDDefaultBufferSize;
 }
 
-// MARK: Public Interface
-@interface DDBufferedProxy<FileLogger: DDFileLogger *> : NSProxy
+@interface DDBufferedProxy : NSProxy
```

Comment:
for what reason?

Reply:
DDBufferedProxy<DDFileLogger *> is not used anywhere, there are no subclasses of DDFileLogger in the repository, and it's not in the header file, so anyone who is subclassing can't actually use the <> syntax. I didn't see it adding much value... thoughts?

Reply:
I already suggested to remove the use of generics [here](https://github.com/CocoaLumberjack/CocoaLumberjack/pull/1001/files/afde4f778ae5b0e0ffb9e8c60b4040cb9b38cfac#r245720727). Again, IMHO this is not necessary at the moment and therefore makes the code more complex than it needs to be. While I gave in in the last discussion about this (because it's a non-breaking change to remove it again), now there are already two people thinking the same way. So I'm with @hhanesand here.

Reply:
@hhanesand @ffried 
I would suggest to add protocol with `lt_logData:` method to emphasize the importance of it in buffer-proxy. It is a hidden power which embodies in temptation to use any other methods of FileLogger and this power shall be chained by rigid types as much as Objective-C offers.

All these new methods just pollute DDFileLogger and make gap between it and any other logger wider.

Reply:
I don't see what should be wrong with the current implementation. We chose `NSProxy` because it can dynamically forward the implementations to `DDFileLogger`. A protocol would be what we should choose if we had two distinct classes. (Or in Swift we could do a lot with protocol extensions, but that's a whole other discussion.)

Reply:
@ffried 
Ah, ok, you win, let's merge it :3